src/main/scala/ch/epfl/bluebrain/nexus/iam/realms/RealmState.scala

```diff
@@ -0,0 +1,17 @@
+package ch.epfl.bluebrain.nexus.iam.realms
+import ch.epfl.bluebrain.nexus.iam.realms.RealmEvent._
+
+sealed trait RealmState extends Product with Serializable
+
+object RealmState {
+
+  final case object Initial                                             extends RealmState
+  final case class Current(realm: Realm, rev: Int, deprecated: Boolean) extends RealmState
```

Comment:
Missing the following information
```scala
def instant: Instant
def identity: Identity
```

Reply:
Probably we also want to store in the current state the `TokenId` and `java.security.PublicKey`, being the tokenId:

```scala
/**
  * A token unique identifier
  *
  * @param iss the token issuer
  * @param kid the KEY ID of the key
  */
final case class TokenId(iss: Uri, kid: String)

object TokenId {
  final implicit val tokenIdDecoder: Decoder[TokenId] = Decoder.forProduct2[TokenId, String, String]("iss", "kid") {
    case (iss, kid) => TokenId(iss, kid)
  }
}
```

You can find [here](https://github.com/BlueBrain/nexus-iam/blob/b2ce115abc1386311471a48a47ae10811bb0cb11/modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/auth/CredentialsStoreActor.scala) more information about it.